src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include "antares/solver/optim-model-filler/ComponentFiller.h"
+
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include "antares/solver/expressions/visitors/EvalVisitor.h"
+#include "antares/solver/optim-model-filler/ReadLinearConstraintVisitor.h"
+#include "antares/study/system-model/variable.h"
+
+namespace Antares::Optimization
+{
+
+ComponentFiller::ComponentFiller(const Study::SystemModel::Component& component):
+    component_(component),
+    evaluationContext_(component_.getParameterValues(), {})
+{
+}
+
+void ComponentFiller::addVariables(Solver::Modeler::Api::ILinearProblem& pb,
+                                   Solver::Modeler::Api::LinearProblemData& data,
+                                   Solver::Modeler::Api::FillContext& ctx)
```

Comment:
Why pass the context here ? Same for constraint and objective

Reply:
not used for now, this will tell the filler how to select the scenario (MC year) and time interval